modules/deprecations.nix

```
@@ -0,0 +1,40 @@
+{lib, ...}: let
+  inherit (builtins) match trace;
+  inherit (lib.asserts) assertMsg;
+  inherit (lib.modules) doRename;
+  inherit (lib.options) showOption;
+  inherit (lib.lists) concatLists;
+
+  mkRenamedOptionModuleUntil = from: to: date: let
+    isValidISODate = dateString: (match "[0-9]{4}-[0-9]{2}-[0-9]{2}" dateString != null);
+  in
+    assert assertMsg (isValidISODate date) "Cannot rename `${showOption from}` to `${showOption to}`: Your date needs to be in ISO 8601 format, ie. yyyy-mm-dd (got ${date} instead).";
+      doRename {
+        inherit from to;
+        visible = false;
+        warn = true;
+        use = trace "Obsolete option `${showOption from}' is used. It was renamed to `${showOption to}', and will be removed on ${date}";
+      };
+in {
+  imports = concatLists [
+    [
+      # removed options
+    ]
+    [
+      # deprecations
+      (mkRenamedOptionModuleUntil
```

What's the point of making it two separate lists exactly? Also, if you're gonna concat two lists, just use ++
```suggestion
  imports = 
    [
      # removed options
    ]
    ++ [
      # deprecations
      (mkRenamedOptionModuleUntil
```